File: implementation/scipamato/scipamato-core-persistence-jooq/src/test/java/ch/difty/scipamato/core/persistence/newsletter/NewsletterTopicFilterConditionMapperTest.java
package ch.difty.scipamato.core.persistence.newsletter;

import static ch.difty.scipamato.core.db.tables.NewsletterTopic.NEWSLETTER_TOPIC;

import ch.difty.scipamato.common.persistence.FilterConditionMapperTest;
import ch.difty.scipamato.common.persistence.GenericFilterConditionMapper;
import ch.difty.scipamato.core.db.tables.NewsletterTopic;
import ch.difty.scipamato.core.db.tables.records.NewsletterTopicRecord;
import ch.difty.scipamato.core.entity.newsletter.NewsletterTopicFilter;

public class NewsletterTopicFilterConditionMapperTest
    extends FilterConditionMapperTest<NewsletterTopicRecord, NewsletterTopic, NewsletterTopicFilter> {

    private final NewsletterTopicFilterConditionMapper mapper = new NewsletterTopicFilterConditionMapper();

    private final NewsletterTopicFilter filter = new NewsletterTopicFilter();

    @Override
    protected NewsletterTopic getTable() {
        return NEWSLETTER_TOPIC;
    }

    @Override
    protected GenericFilterConditionMapper<NewsletterTopicFilter> getMapper() {
        return mapper;
    }

    @Override
    protected NewsletterTopicFilter getFilter() {
        return filter;
    }
}

File: implementation/scipamato/core/persistence-jooq/src/test/kotlin/ch/difty/scipamato/core/persistence/newsletter/NewsletterTopicFilterConditionMapperTest.kt
package ch.difty.scipamato.core.persistence.newsletter

import ch.difty.scipamato.common.persistence.FilterConditionMapperTest
import ch.difty.scipamato.core.db.tables.NewsletterTopic
import ch.difty.scipamato.core.db.tables.NewsletterTopic.NEWSLETTER_TOPIC
import ch.difty.scipamato.core.db.tables.records.NewsletterTopicRecord
import ch.difty.scipamato.core.entity.newsletter.NewsletterTopicFilter

internal class NewsletterTopicFilterConditionMapperTest : FilterConditionMapperTest<NewsletterTopicRecord, NewsletterTopic, NewsletterTopicFilter>() {
    override val mapper = NewsletterTopicFilterConditionMapper()
    override val filter = NewsletterTopicFilter()
    override val table: NewsletterTopic = NEWSLETTER_TOPIC
}